test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -5230,15 +5276,63 @@ public MockSiteData register(SiteAndWebExchange sawe) {
     public final String strListID;
     public final String strItemID;
 
-    public URLSegmentsExchange(String strURL, boolean getURLSegmentsResult,
-        String strWebID, String strBucketID, String strListID,
-        String strItemID) {
-      this.strURL = strURL;
-      this.getURLSegmentsResult = getURLSegmentsResult;
-      this.strWebID = strWebID;
-      this.strBucketID = strBucketID;
-      this.strListID = strListID;
-      this.strItemID = strItemID;
+    public URLSegmentsExchange(Builder builder) {
+      this.strURL = builder.strURL;
+      this.getURLSegmentsResult = builder.getURLSegmentsResult;
+      this.strWebID = builder.strWebID;
+      this.strBucketID = builder.strBucketID;
+      this.strListID = builder.strListID;
+      this.strItemID = builder.strItemID;
+    }
+
+    private static class Builder {
+      private String strURL;
+      private boolean getURLSegmentsResult;
+      private String strWebID;
+      private String strBucketID;
+      private String strListID;
+      private String strItemID;
+
+      static Builder fromURLSegmentsExchange(URLSegmentsExchange base) {
+        return new Builder().setStrURL(base.strURL)
+            .setGetURLSegmentsResult(base.getURLSegmentsResult)
+            .setStrWebID(base.strWebID).setStrBucketID(base.strBucketID)
+            .setStrListID(base.strListID).setStrItemID(base.strItemID);
+      }
+
+      Builder setStrURL(String strURL) {
```

Comment:
What about making the URL a constructor argument, since it appears to be required, and could even use this.strURL = Preconditions.checkNotNull(strURL)? And the same for the object type in ContentExchange.Builder?

Reply:
Done